README.adoc

```diff
@@ -0,0 +1,161 @@
+
+=== Infrastructure
+
+* A Docker repository that the Kubernetes cluster has the necessary credentials to pull from. You can get a free Docker registry at https://hub.docker.com/
+
+For the deployment, there are two possible choices. Below we highlight the main characteristic of each deployment approach. Further details about prerequisites and infrastructure preparation would be discussed further in the <<deployment,Deployment>> section. 
+
+==== Deploy on Lighbend Cloudstate managed service
+
+Basically, all you need is to open an account with Lightbend Cloudstate hosting service. Then install the "Cloudstate CLI" tool and you are good to deploy. There is no need to provide any local infrastructure.
+
+The deployment is as simple as applying the yaml configuration files. Lightbend Cloudstate managed service will take care of the hosting, configuration and maintenance of your service. 
+
+==== Deploy on your own Kubernetes
+
+In case you prefer to host your own services. You need to provide a Kubernetes cluster. This cluster could be local or cloud-based and must fulfill the following requirements:
+
+* A Kubernetes cluster, with 1.13 as the minimum required version
+* An ingress controller with a Layer7 Load Balancer ability
+* A working knowledge of Kubernetes to be able to manage your cluster and to troubleshoot, should any issue specific to your Kubernetes cluster arise.
+
+
+== Build Docker images
+
+Before beginning the deployment, the various microservices composing the chat application must be packaged each into their own docker image, and published to your docker registry.
+
+=== Building the Friends & Presence services
+* https://github.com/cloudstateio/samples-js-chat/blob/master/friends/README.adoc#building-the-friends-service[Building the Friends service]
+
+* https://github.com/cloudstateio/samples-js-chat/blob/master/presence/README.adoc#building-the-presence-service[Building the Presence service]
+
+=== Building the Chat-UI service
+
+The code below assumes that your docker registry is named `mydockerregistry`. Please replace it with your DockerID (which you have registed at https://hub.docker.com/ as mentioned in <<infrastructure>>
+
+
+[source,shell]
+----
+cd ./chat-ui
+
+nvm install
+nvm use
+npm install
+./protogen.sh
+npm run prestart
+npm run-script build
+
+# let's assume your DockerID in https://hub.docker.com/ is `mydockerregistry`
+DOCKER_PUBLISH_TO=mydockerregistry
+
+# build docker image
+docker build . -t $DOCKER_PUBLISH_TO/samples-js-chat-ui:latest
+
+# authenticate with your Docker registry
+docker login
+
+# push the docker image to your registry
+docker push $DOCKER_PUBLISH_TO/samples-js-chat-ui:latest
+----
+
+=== Testing the Chat-UI service
+
+In order to make the deployment of the Chat UI consistent with all the other services composing the Chat application. The Node.js server running the UI code is wrapped in a Cloudstate stateless service.
```

Comment:
"Chat application" is named inconsistently within phrases, sometimes with "**C**hat application" and then as "**c**hat application"